### DISCUSSION TITLE
怎么基于本地的第三方插件进行二次开发

### DISCUSSION THREAD

**Discussion by nidongpinyinme** [obsidianzh/forum#26] [Q&A常见问题] [answered]
- 从软件下载的插件文件中一般包含了“manifest.json”和“main.js“两个文件，我对其中的”main.js“进行了修改，然后应该怎么让obsidian中的插件更新。
- 或者说我应该怎么进行二次开发。
- 顺便问一下大家都是怎么调试的。

**Comment by makerjackie** [comment, marked as answer]
复制粘贴一下，来源：https://forum.obsidian.md/t/plugins-mini-faq/7737/6

I want to develop a plugin, where do I start?
Prerequisites: Javascript, Typescript, npm, git knowledge as far as I can tell
* Where is the API? https://github.com/obsidianmd/obsidian-api 54
* How should a plugin look like? https://github.com/obsidianmd/obsidian-sample-plugin 37
The README provides additional instructions on how to use, how to create releases, and how to register the plugin in the community plugin page.
You can use the sample plugin as a template for your own plugin

你可以到[obsidian的官方英文论坛的developer频道](https://forum.obsidian.md/c/developers-api/14)中去看看大家的讨论。

有空我也学学typescripts来开发一下好玩的插件

**Reply by Quorafind** [reply]
期待

**Reply by nidongpinyinme** [reply]
谢谢老哥的链接。
同时自问自答一下：我修改了main.js后**重启**了obsidian，发现我的改动生效了。

**Reply by ericaxu** [reply]
欢迎二次开发，不过最好单独放一个文件夹，否则更新的时候会直接被覆盖哦。
